src/main/scala/org/broadinstitute/dsde/firecloud/webservice/ManagedGroupApiService.scala

```diff
@@ -0,0 +1,76 @@
+import org.broadinstitute.dsde.workbench.model.{WorkbenchEmail, WorkbenchGroupName}
+import org.slf4j.LoggerFactory
+import spray.http.HttpMethods
+import spray.routing._
+
+trait ManagedGroupApiService extends HttpService with PerRequestCreator with FireCloudRequestBuilding with FireCloudDirectives with StandardUserInfoDirectives {
+
+  private implicit val executionContext = actorRefFactory.dispatcher
+
+  lazy val log = LoggerFactory.getLogger(getClass)
+
+  val managedGroupServiceConstructor: (WithAccessToken) => ManagedGroupService
+
+  val managedGroupServiceRoutes =
+    pathPrefix("api") {
+      pathPrefix("groups") {
+        pathEnd {
+          get {
+            requireUserInfo() { userInfo =>
+              requestContext =>
+                perRequest(requestContext, ManagedGroupService.props(managedGroupServiceConstructor, userInfo), ListGroups)
+            }
+          }
+        } ~
+        pathPrefix(Segment) { groupName =>
+          pathEnd {
+            get {
+              requireUserInfo() { userInfo =>
+                requestContext =>
+                  perRequest(requestContext, ManagedGroupService.props(managedGroupServiceConstructor, userInfo), ListGroupMembers(WorkbenchGroupName(groupName)))
+              }
+            } ~
+            post {
+              requireUserInfo() { userInfo =>
+                requestContext =>
+                  perRequest(requestContext, ManagedGroupService.props(managedGroupServiceConstructor, userInfo), CreateGroup(WorkbenchGroupName(groupName)))
+              }
+            } ~
+            delete {
+              requireUserInfo() { userInfo =>
+                requestContext =>
+                  perRequest(requestContext, ManagedGroupService.props(managedGroupServiceConstructor, userInfo), DeleteGroup(WorkbenchGroupName(groupName)))
+              }
+            }
+          } ~
+          //must be re-implemented in sam (GAWB-3335)
+//          path("requestAccess") {
+//            post {
+//              passthrough(UserApiService.rawlsGroupRequestAccessUrl(groupName), HttpMethods.POST)
+//            }
+//          } ~
+          path(Segment / Segment) { (role, email) =>
+            put {
+              requireUserInfo() { userInfo =>
+                requestContext =>
+                  perRequest(requestContext, ManagedGroupService.props(managedGroupServiceConstructor, userInfo), AddGroupMember(WorkbenchGroupName(groupName), ManagedGroupRoles.withName(role), WorkbenchEmail(email)))
+              }
+            } ~
+            delete {
+              requireUserInfo() { userInfo =>
+                requestContext =>
+                  perRequest(requestContext, ManagedGroupService.props(managedGroupServiceConstructor, userInfo), RemoveGroupMember(WorkbenchGroupName(groupName), ManagedGroupRoles.withName(role), WorkbenchEmail(email)))
+              }
+            }
+          }
+        }
+      }
+    }
+}
```

Comment:
Why does Orch need these endpoints?  How come things can't talk directly to SAM?

Reply:
everything in the firecloud UI goes through orch at the moment, which i think is mostly a leftover from how things used to work in the past (none of the underlying services were exposed to the public, only thru orch). ideally these could just be passthroughs (like they previously were) but because a lot of the responses changed, i had to implement extra logic in here to preserve as much backwards compatibility as possible.

saturn UI will go to the individual services themselves which will be 🔥

Reply:
Should we mark all of this Managed Group stuff in Orch as deprecated?  Looks like we could do this in swagger in addition to the code.